addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/handler/SolarEdgeBaseHandler.java

```diff
@@ -0,0 +1,201 @@
+        SolarEdgeConfiguration config = getConfiguration();
+
+        logger.debug("Solaredge initialized with configuration: {}", config);
+
+        this.webInterface = new WebInterface(config, this);
+
+        if (config.getPassword() != null && config.getUsername() != null) {
+            this.startPolling();
+        } else {
+            thing.setStatusInfo(new ThingStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "no username/password set"));
+        }
+    }
+
+    /**
+     * Start the polling.
+     */
+    private synchronized void startPolling() {
+        if (liveDataPollingJob == null || liveDataPollingJob.isCancelled()) {
+            logger.debug("start live data polling job at intervall {}",
+                    getConfiguration().getLiveDataPollingInterval());
+            liveDataPollingJob = scheduler.scheduleWithFixedDelay(liveDataPollingRunnable, 1,
+                    getConfiguration().getLiveDataPollingInterval(), TimeUnit.SECONDS);
+        } else {
+            logger.debug("live data pollingJob already active");
+        }
+        if (aggregateDataPollingJob == null || aggregateDataPollingJob.isCancelled()) {
+            logger.debug("start aggregate data polling job at intervall {}",
+                    getConfiguration().getAggregateDataPollingInterval());
+            liveDataPollingJob = scheduler.scheduleWithFixedDelay(aggregateDataPollingRunnable, 1,
+                    getConfiguration().getAggregateDataPollingInterval(), TimeUnit.SECONDS);
+        } else {
+            logger.debug("aggregate data pollingJob already active");
+        }
+    }
+
+    /**
+     * Disposes the bridge.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        if (liveDataPollingJob != null && !liveDataPollingJob.isCancelled()) {
+            logger.debug("stop live data polling job");
+            liveDataPollingJob.cancel(true);
+            liveDataPollingJob = null;
+        }
+        if (aggregateDataPollingJob != null && !aggregateDataPollingJob.isCancelled()) {
+            logger.debug("stop aggregate data polling job");
+            aggregateDataPollingJob.cancel(true);
+            aggregateDataPollingJob = null;
+        }
+    }
+
+    @Override
+    public WebInterface getWebInterface() {
+        return webInterface;
+    }
+
+    /**
+     * will update all channels provided in the map
+     */
+    @Override
+    public void updateChannelStatus(Map<String, String> values) {
+        logger.debug("Handling channel update.");
+
+        for (String key : values.keySet()) {
+            Channel channel = getThingSpecificChannel(key);
+            if (channel != null) {
+                String value = values.get(key);
+                logger.debug("Channel is to be updated: {}: {}", channel.getFQName(), value);
+                if (value != null) {
+                    if (channel.getJavaType().equals(Double.class)) {
```

Comment:
This looks non-standard to me, afaik there should normally be no magic depending on a mapped type, @kaikreuzer wdyt?

Reply:
Channels can be String or number. I used this code to decide which type should be used in Java. Of course I could map everything to string. But this is also not a good idea.
Another option would be to assume that everything which cannot be converted to a number will be a string. But this is much more magic because that decision is then made on information which is only available on runtime. My solution is at least static.

Reply:
@kaikreuzer your help is appreciated here :-)

Reply:
@martinvw I think the confusion comes from the fact that `Channel` is actually not the ESH class here, but a binding specific class. So it's ok for me to use that as binding internal data in order to decide what should be sent.

Reply:
@martinvw : This is now refactored due to the UoM integration